### PR TITLE
Add methods on Txbuilder

### PR DESCRIPTION
**Pull request by thunderbiscuit**
This PR adds a few new methods on the TxBuilder.

Todo:
- [x] tests
- [x] Open up BumpFeeTxBuilder

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:
* [x] I've added tests for the new feature

### CODE REVIEW

File: bdk-swift/Tests/BitcoinDevKitTests/LiveTxBuilderTests.swift
Comment (reez) [inline bdk-swift/Tests/BitcoinDevKitTests/LiveTxBuilderTests.swift:1]: No idea why this code shows up as grey in the GitHub UI here? Regardless it looks good to me locally

**Review by reez — APPROVED** [review]
ACK bbc6e1a43ca91662fe00ea6dcbe51ba87ae46be6

Reviewed, tested locally, nice job w the tests across all 4 platforms.

File: bdk-swift/Tests/BitcoinDevKitTests/LiveTxBuilderTests.swift
Comment (thunderbiscuit) [reply bdk-swift/Tests/BitcoinDevKitTests/LiveTxBuilderTests.swift:1]: Oh weird... no I haven't seen this before.

Comment (thunderbiscuit) [reply bdk-swift/Tests/BitcoinDevKitTests/LiveTxBuilderTests.swift:1]: Aaah I think it's related to the `/*` used in the descriptor on line 7. Is that maybe a long-form comment that the IDE interprets as never closed?

**Pull request merged**